consensus/src/state_computer.rs

```diff
@@ -42,6 +43,7 @@ pub struct ExecutionProxy {
     async_state_sync_notifier: channel::Sender<NotificationType>,
     async_commit_notifier: channel::Sender<CommitType>,
     validators: Mutex<Vec<AccountAddress>>,
+    write_mutex: AsyncMutex<()>,
```

Comment:
Why don't we create a mutex around the executor? we are accessing executor only with mutex locked?

Reply:
we need to be able to execute and commit concurrently.